src/Widgets/DataFormContainerWidget/DataFormContainerWidgetComponent.tsx

```diff
@@ -84,17 +86,12 @@ provideComponent(DataFormContainerWidget, ({ widget }) => {
     setKeyCounter((k) => k + 1)
   }
 
-  function toastAndRedirect(targetDataItem: DataItem) {
+  async function toastAndRedirect(targetDataItem: DataItem) {
     if (submittedMessage) toast.success(submittedMessage)
 
     if (redirectAfterSubmit) {
-      // TODO: Remove this work around once #10212 is resolved
-      navigateTo(redirectAfterSubmit, {
-        params: {
-          [`${snakeCase(targetDataItem.dataClass().name())}_id`]:
-            targetDataItem.id(),
-        },
-      })
+      const url = await load(() => urlForDataItem(targetDataItem))
+      if (url) navigateTo(new Link({ url }))
```

Comment:
* [ ] FYI The indirection through `new Link` looks like a workaround. We could use `if (url) getHistory()?.push(url)` instead.

Reply:
This is a workaround for issue #10629 in the SDK.

Reply:
😆 So my impression was correct. Using history is better as it's not a workaround but a valid alternative implementation in the light of a missing convenience feature.

Reply:
But history "destroys" the current app state, if I'm not mistaken. So all objs and data needs to be loaded again. With `navigateTo` it is more seamless.

Reply:
> But history "destroys" the current app state

No, see https://github.com/Scrivito/scrivito-portal-app/blob/d696a2077b4c27610f2cce31fbf65f7ed33d199b/src/config/history.ts#L9

Reply:
Good point!